include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -450,6 +443,11 @@ void multiplexer::process_peek(boost::system::error_code error,
     header::data_type header_data;
     std::size_t payload_size = datagram_size - header_size;
 
+    // Temporarily increase the count by one to prevent callbacks
+    // from decreasing it to zero and thus sending ourself an empty packet.
+    // It shall be decreased back by one at the end of this function.
+    ++receive_calls;
```

Comment:
I think they can in one of the process_{handshake,data} functions as they call user's callbacks.